### PR TITLE
St 222 button refactoring

### PR DESCRIPTION
**Pull request by bigKingPenguin**

### CODE REVIEW

File: frontend/components/common/Button.vue [hunk 1]
```diff
@@ -1,123 +1,263 @@
...
+		:is="link ? NuxtLink : 'button'"
+		:type="link ? null : 'button'"
+		:to="link ? link : null"
+		:target="isExternalLink ? '_blank' : null"
+		:disabled="!link && isDisabled"
+		:class="[
+			isIcon ? 'icon' : `button button__${buttonKind}`,
+			isIcon && buttonKind ? `icon__${buttonKind}` : '',
+			isDisabled ? `button__${buttonKind}--disabled` : '',
+			isRound ? 'button--round' : ''
```
Comment (Bambukoz) [inline frontend/components/common/Button.vue:84]: {'button--round': isRound}

File: frontend/components/modal/NeedAuthorize.vue [hunk 2]
```diff
@@ -22,7 +22,7 @@ const props = defineProps<Props>();
 			</div>
 			<div class="modal-card__foot">
 				<CommonButton
-					button-class="button__success"
+					button-kind="success"
 					button-text="Закрыть"
```
Comment (Bambukoz) [inline frontend/components/modal/NeedAuthorize.vue:26]: перевод

File: frontend/pages/event/[id].vue [hunk 3]
```diff
@@ -106,37 +103,41 @@ const templateURL = computed(
...
+			<div
+				v-if="user?.id === posterEvent.creatorId"
+				class="event-actions__manage"
+			>
 				<CommonButton
 					class="event-actions__button"
-					button-class="button__success"
 					:button-text="$translate('event.button.edit')"
+					icon-name="edit"
 					@click="openEventModal()"
```
Comment (Bambukoz) [inline frontend/pages/event/[id].vue:130]: openEventModal без ()

Comment (bigKingPenguin) [reply frontend/pages/event/[id].vue:130]: +

File: frontend/pages/event/[id].vue [hunk 4]
```diff
@@ -106,37 +103,41 @@ const templateURL = computed(
...
+					is-external-link
 				/>
 
 				<CommonButton
 					v-else
+					button-kind="success"
 					class="event-actions__button"
-					button-class="button__success"
 					:button-text="$translate('event.button.register')"
 					@click="openRegistrationModal()"
```
[the remark this answers is not in the document]
Comment (bigKingPenguin) [reply frontend/pages/event/[id].vue:118]: +

File: frontend/components/modal/Registration.vue [hunk 5]
```diff
@@ -206,13 +206,13 @@ const closeModal = () => {
...
+					button-kind="ordinary"
 					:button-text="$translate('component.new_event_modal.cancel')"
 					@click="closeModal()"
 				/>
 				<CommonButton
 					class="modal-card__button"
-					button-class="button__success"
+					button-kind="success"
 					:button-text="$translate('component.new_event_modal.submit')"
 					@click="submit()"
```
[the remark this answers is not in the document]
Comment (bigKingPenguin) [reply frontend/components/modal/Registration.vue:217]: +

File: frontend/components/modal/Registration.vue [hunk 6]
```diff
@@ -206,13 +206,13 @@ const closeModal = () => {
 			<div class="modal-card__foot">
 				<CommonButton
 					class="modal-card__button"
-					button-class="button__ordinary"
+					button-kind="ordinary"
 					:button-text="$translate('component.new_event_modal.cancel')"
 					@click="closeModal()"
```
[the remark this answers is not in the document]
Comment (bigKingPenguin) [reply frontend/components/modal/Registration.vue:211]: +

File: frontend/components/modal/NeedAuthorize.vue (hunk 2 above)
Comment (bigKingPenguin) [reply frontend/components/modal/NeedAuthorize.vue:26]: done

File: frontend/components/common/Button.vue [hunk 7]
```diff
@@ -1,123 +1,263 @@
...
+		@click="!link && !isDisabled ? emit('click') : null"
+	>
+		<CommonUiLoadSpinner
+			v-if="isLoading"
+			:color="loaderColor"
+			class="loader"
+		/>
+		<CommonIcon
+			v-if="iconName"
+			:class="buttonText ? 'button__icon' : ''"
```
[the remark this answers is not in the document]
Comment (bigKingPenguin) [reply frontend/components/common/Button.vue:96]: done

File: frontend/components/common/Button.vue [hunk 8]
```diff
@@ -1,123 +1,263 @@
...
+		type: String as PropType<ButtonKind>,
+		default: 'ordinary'
+	},
+	buttonText: {
+		type: String as PropType<string>,
+		default: ''
+	},
+	link: {
+		// если это ссылка
+		type: [String, Object] as PropType<string | { name?: string; path?: string }>,
```
[the remark this answers is not in the document]
Comment (bigKingPenguin) [reply frontend/components/common/Button.vue:21]: done

**Review by Bambukoz — APPROVED** [review]

**Comment by bigKingPenguin** [comment]
#222

**Pull request merged**